compiler/src/dmd/backend/x86/cod2.d

```diff
@@ -4664,7 +4664,7 @@ void cdrelconst(ref CGstate cg, ref CodeBuilder cdb,elem *e,ref regm_t pretregs)
         case TYildouble:
         case TYcldouble:
             tym = TYnptr;               // don't confuse allocreg()
-            if (pretregs & (mES | mCX) || e.Ety & mTYfar)
+            if (I16 && pretregs & (mES | mCX) || e.Ety & mTYfar)
```

Comment:
Looks like this is unrelated?

Reply:
yes, I was going to make it a separate PR. It's a problem that's been there forever, but it doesn't seem to negatively affect code execution

Reply:
https://github.com/dlang/dmd/pull/17065